sympy/matrices/expressions/matadd.py

```diff
@@ -51,7 +51,7 @@ def _eval_adjoint(self):
 
     def _eval_trace(self):
         from trace import Trace
-        return MatAdd(*[Trace(arg) for arg in self.args]).doit()
+        return Add(*[Trace(arg).doit() for arg in self.args])
```

Comment:
Wait, maybe I stand by it not evaluating... but test might be wrong in current form.  Will think about it more...